src/analytics/SegmentAnalyticsService.js

```diff
@@ -187,13 +192,7 @@ class SegmentAnalyticsService {
         resolve();
       });
 
-      // if the segment domain is blocked on user's network/browser, this promise does not get
-      // resolved and user see a blank page, set timeout out to resolve the promise.
-      setTimeout(() => {
-        // Other segment calls won't work if we are not identified. So let's just un-initialize ourselves.
-        this.segmentInitialized = false;
-        resolve();
-      }, 2000);
+      document.addEventListener('segmentFailed', resolve);
```

Comment:
I would expect the segmentFailed event to fire before this listener is added.  Is it the case that you only expect this to catch the segmentFailed event if the script loading has timed out, and thus is likely to occur after identifyAnonymousUser is called?

If so, I'd add some comments to that effect, explaining that it'll only be used in certain circumstances.  Otherwise someone might delete the code thinking it doesn't function as written.

Reply:
@davidjoy that was my concern as well with this but when I blocked the segment CDN domain in my browser during my testing it always worked as expected resolved the promise. And this is generally for all error cases but specifically for the blocking scenario instead of timeout.

Sure, I'll add the explaining comment.

Reply:
@davidjoy added the comment, kindly take another look.